src/test/java/com/microsoft/azure/functions/worker/binding/tests/RpcHttpRequestDataSourceTest.java

```diff
@@ -41,14 +56,150 @@ public static RpcHttp getTestRpcHttp(Object inputBody) throws Exception {
+    JavaFunctionBroker broker = new JavaFunctionBroker(classLoader);
+    FunctionEnvironmentReloadRequestHandler envHandler = new FunctionEnvironmentReloadRequestHandler(broker);
+    Map<String, String> existingVariables = System.getenv();
+    Map<String, String> newEnvVariables = new HashMap<>();
+    newEnvVariables.putAll(existingVariables);
+    newEnvVariables.put("FUNCTIONS_WORKER_NULLABLE_VALUES_ENABLED", "true");
+    envHandler.setEnv(newEnvVariables);
+    Method httpRequestMessageStringBodyMethod = getFunctionMethod("HttpRequestStringBody");
+    Map<String, String> queryMap = new HashMap<String, String>() {{
+      put("name", "");
+    }};
+    Parameter[] parameters = httpRequestMessageStringBodyMethod.getParameters();
+    String sourceKey = "testRpcHttp";
+    RpcHttp input = getTestRpcHttp(null, null, queryMap);
+    RpcHttpRequestDataSource rpcHttp = new RpcHttpRequestDataSource(sourceKey, input);
+    Optional<BindingData> actualBindingData = rpcHttp.computeByName(sourceKey,
+            parameters[0].getParameterizedType());
+    BindingData actualArg = actualBindingData.orElseThrow(WrongMethodTypeException::new);
+    HttpRequestMessage<?> requestMsg = (HttpRequestMessage<?>) actualArg.getValue();
+    assertEquals(requestMsg.getQueryParameters().get("name"), "");
+  }
+
+  @Test
+  public void rpcHttpDataSource_To_HttpRequestMessage_NullableQueryParamsEmpty_EnvSettingDisabled() throws Exception {
+    DefaultClassLoaderProvider classLoader = new DefaultClassLoaderProvider();
+    JavaFunctionBroker broker = new JavaFunctionBroker(classLoader);
+    FunctionEnvironmentReloadRequestHandler envHandler = new FunctionEnvironmentReloadRequestHandler(broker);
+    Map<String, String> existingVariables = System.getenv();
+    Map<String, String> newEnvVariables = new HashMap<>();
+    newEnvVariables.putAll(existingVariables);
+    newEnvVariables.put("FUNCTIONS_WORKER_NULLABLE_VALUES_ENABLED", "false");
+    envHandler.setEnv(newEnvVariables);
+    Method httpRequestMessageStringBodyMethod = getFunctionMethod("HttpRequestStringBody");
+    Map<String, String> queryMap = new HashMap<String, String>() {{
+      put("name", "");
+    }};
+    Parameter[] parameters = httpRequestMessageStringBodyMethod.getParameters();
+    String sourceKey = "testRpcHttp";
+    RpcHttp input = getTestRpcHttp(null, null, queryMap);
+    RpcHttpRequestDataSource rpcHttp = new RpcHttpRequestDataSource(sourceKey, input);
+    Optional<BindingData> actualBindingData = rpcHttp.computeByName(sourceKey,
+            parameters[0].getParameterizedType());
+    BindingData actualArg = actualBindingData.orElseThrow(WrongMethodTypeException::new);
+    HttpRequestMessage<?> requestMsg = (HttpRequestMessage<?>) actualArg.getValue();
+    assertEquals(requestMsg.getQueryParameters().get("name"), null);
+  }
+
+  @Test
+  public void rpcHttpDataSource_To_HttpRequestMessage_NullableQueryParamsNonEmpty() throws Exception {
```

Comment:
I think we can have the test cases more clear. There are two major scenarios we want to test
1. appsetting to false
2. appsettting to true

each of those scenarios we want to test
1. normal case -- param value empty
2. param value empty

So we can have two test cases in total
1. with appsetting to true

- have one normal param -- expect same value
- have one param with empty value -- expect empty string
2. with appsetting to false

- have one normal param -- expect normal value
- have one param with empty value -- expect null

We can test header map and query map in same test method. 
So we can reduce the test methods number to 2 but still have full coverage. Let me know if this make sense or not. Thanks.